### DISCUSSION TITLE
Elsa breaking changes

### DISCUSSION THREAD

**Discussion by Alexplose** [elsa-workflows/elsa-core#1554] [Q&A]
I moved from Elsa 1.5 to 2.X, there was a lot of breaking changes.

Since we start to use it more and more, do you plan to give update path so that breaking changes in future version are less painful ?

Thanks

**Comment by sfmskywalker** [comment]
Hi @Alexplose, sorry about that. I do hope to make upgrades less painful going forward, but realistically I may not be able to pull this of until after 3.0, which will be another major change compared to 2.0, in much the same way that 2.0 was a major change compared to 1.0.

I of course fully realize the problem with this for systems that heavily rely on Elsa, which is why we're taking steps with 3.0 to design an "Elsa" standard not unlike the BPMN standard. Our hope is that with such a standard, at the very least the workflows won't break when going from 3.0 to 4.0, even if the underlying code APIs might break a little from version to version.

Because of that effort, it might take a while before 3.0 sees the day of light. Which means until then, 2.x will see a few more updates before it's pronounced "done". 

Hopefully this will allow you to take advantage of the 2.0 brand for a good amount of time, and perhaps stick with it even after 3.0 has shipped until you find reason enough to go through another migration from 2.x to 3.0 (which may not even happen because 2.x is getting better every day).
